src/vs/workbench/api/node/mainThreadLanguageFeatures.ts

```diff
@@ -21,13 +21,15 @@ import { LanguageConfigurationRegistry } from 'vs/editor/common/modes/languageCo
 import { LanguageConfiguration } from 'vs/editor/common/modes/languageConfiguration';
 import { IHeapService } from './mainThreadHeapService';
 import { IModeService } from 'vs/editor/common/services/modeService';
+import { MainThreadHandlerRegistry } from 'vs/workbench/api/node/mainThreadHandlerRegistry';
 
 export class MainThreadLanguageFeatures extends MainThreadLanguageFeaturesShape {
 
 	private _proxy: ExtHostLanguageFeaturesShape;
 	private _heapService: IHeapService;
 	private _modeService: IModeService;
 	private _registrations: { [handle: number]: IDisposable; } = Object.create(null);
+	private _progressRegistrations = new MainThreadHandlerRegistry<(value: any) => void>();
```

Comment:
Correct, but I am opened to narrowing the type for now. I will add a doc comment here.